### DISCUSSION TITLE
EZ Landing

### DISCUSSION THREAD

**Discussion by TelefraggerRick** [iNavFlight/inav#10189] [Ideas]
Wondering if we could get a ez landing mode like betaflight to make acro landings with a multirotor a little more smooth. Basically disabling airmode with sticks at near neutral position and no throttle. Would only really be needed for acro mode.

"This is a newly developed feature, CLI only, that makes landings less bouncy, even when airmode is on. This is achieved by restricting the amount to which airmode can increase throttle, and by attenuating iTerm, when throttle is low and sticks are centred.

EzLanding is disabled by default.

    To enable, go set mixer_type = EZLANDING in CLI.
    To return to normal behaviour, go set mixer_type = LEGACY in CLI.

There are two tuning parameters:

    ez_landing_limit: Default: 5, Range: 0-75. Allowed maximum percentage throttle increase via airmode, with sticks centered and throttle at zero. Higher values provide a bit more stability when perching or in flat drops. Lower values make landings less bouncy.
    ez_landing_threshold: Default: 25, Range: 0-200. Percentage stick deflection at which airmode is given full authority to adjust the throttle, with linear attenuation towards the center

The EzLanding effect is strongest when the sticks are centred and throttle is at zero. Under these conditions there will be a small reduction in PID stabilisation. To retain a bit more stability, eg when trying to 'perch' on an object, or during flat or inverted zero throttle drops, retain a tiny bit of throttle during the move.

For more information, see [PR 12094](https://github.com/betaflight/betaflight/pull/12094). Debug: set debug_mode = EZLANDING"

**Comment by breadoven** [comment]
If you set `nav_disarm_on_landing `to OFF on a multirotor you get a similar affect with PID accumulators reset when landing is detected with the throttle held low. Admittedly this will prevent disarm during automated flight. However, it's always possible to introduce different disarm behaviour for automated and Acro modes.
